vars/buildTestContainer.groovy

```diff
@@ -0,0 +1,74 @@
+/**
+ * Build/Test/Push container to a DockerHub
+ * @param parameters
+ * versions: A list of versions to tag the image with. If supplied, the version will be pushed to docker hub
+ * image_name: The name of the image that should be pushed to docker hub
+ * credentials: Credentials for container push. Must contain USERNAME, PASSWORD as variables
+ * @return
+ */
+
+def call(Map parameters = [:]) {
+    def versions = parameters.versions ?: []
+    def build_cmd = parameters.build_cmd ?: 'build'
+    def test_cmd = parameters.test_cmd ?: 'test'
+    def image_name = parameters.image_name
+    def container_registry = parameters.container_registry ?: 'docker://docker.io'
+    def container_namespace = parameters.container_namespace
+    def buildContainer = parameters.buildContainer ?: 'buildah'
+    def credentials = parameters.credentials ?: []
+    def build_root = parameters.build_root ?: '.'
```

Comment:
nit: would we need params for tag and port# for future use-cases?

Reply:
Do you have an example use-case? Docker tags are passed in through the versions parameter.